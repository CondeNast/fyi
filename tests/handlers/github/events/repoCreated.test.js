const { Application } = require('probot')
const plugin = require('../../../../robot')
const repoCreatedEvent = require('../../../fixtures/repo-created')
const models = require('../../../../models')
const Event = models.Event

describe('Arch Bot', () => {
  afterAll(async () => {
    await models.sequelize.close()
  })

  describe('Repository Created', () => {
    it('Creates an Issue in Admin Repository', async () => {
      await app.receive(repoCreatedEvent)
      expect(github.issues.create).toMatchSnapshot()
      expect(Event.create).toMatchSnapshot()
    })
  })
})
