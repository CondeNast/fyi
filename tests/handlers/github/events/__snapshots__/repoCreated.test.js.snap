// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arch Bot Repository Created Creates an Issue in Admin Repository 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "assignees": Array [
          "johnkpaul",
          "gautamarora",
        ],
        "body": "Repository Name: <a href=\\"https://github.com/choosenearme/my-awesome-project\\">choosenearme/my-awesome-project</a>
Created By: <a href=\\"https://github.com/awesome-coder\\">awesome-coder</a>

<!-- probot = {\\"6732\\":{\\"type\\":\\"fyi\\",\\"org\\":\\"choosenearme\\",\\"repo\\":\\"my-awesome-project\\",\\"repoCreator\\":\\"awesome-coder\\"}} -->
",
        "labels": Array [
          "repo-created",
        ],
        "number": undefined,
        "owner": "choosenearme",
        "repo": "fyis",
        "title": "Repo Created: choosenearme/my-awesome-project",
      },
    ],
  ],
}
`;
